fastly/tls_mutual_authentication.go

```diff
@@ -0,0 +1,212 @@
+	CreatedAt   *time.Time       `jsonapi:"attr,created_at,iso8601"`
+	Enforced    bool             `jsonapi:"attr,enforced"`
+	ID          string           `jsonapi:"primary,mutual_authentication"`
+	Name        string           `jsonapi:"attr,name"`
+	UpdatedAt   *time.Time       `jsonapi:"attr,updated_at,iso8601"`
+}
+
+// ListTLSMutualAuthenticationsInput is used as input to the Client.ListTLSMutualAuthentication function.
+type ListTLSMutualAuthenticationsInput struct {
+	// Include is a comma-separated list of related objects to include (optional).
+	Include string
+	// PageNumber is the required page index for pagination.
+	PageNumber int
+	// PageSize is the number of records per page.
+	PageSize int
+}
+
+// formatFilters converts user input into query parameters for filtering.
+func (i *ListTLSMutualAuthenticationsInput) formatFilters() map[string]string {
+	result := map[string]string{}
+	pairings := map[string]interface{}{
+		"include":      i.Include,
+		"page[size]":   i.PageSize,
+		"page[number]": i.PageNumber,
+	}
+
+	for key, value := range pairings {
+		switch t := reflect.TypeOf(value).String(); t {
+		case "string":
+			if value != "" {
+				result[key] = value.(string)
+			}
+		case "int":
+			if value != 0 {
+				result[key] = strconv.Itoa(value.(int))
+			}
+		}
+	}
+
+	return result
+}
+
+// ListTLSMutualAuthentication retrieves all resources.
+func (c *Client) ListTLSMutualAuthentication(i *ListTLSMutualAuthenticationsInput) ([]*TLSMutualAuthentication, error) {
+	p := "/tls/mutual_authentications"
+	filters := &RequestOptions{
+		Params: i.formatFilters(),
+		Headers: map[string]string{
+			"Accept": "application/vnd.api+json", // this is required otherwise the filters don't work
+		},
+	}
+
+	r, err := c.Get(p, filters)
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := jsonapi.UnmarshalManyPayload(r.Body, reflect.TypeOf(new(TLSMutualAuthentication)))
+	if err != nil {
+		return nil, err
+	}
+
+	so := make([]*TLSMutualAuthentication, len(data))
+	for i := range data {
+		typed, ok := data[i].(*TLSMutualAuthentication)
+		if !ok {
+			return nil, fmt.Errorf("unexpected response type: %T", data[i])
+		}
+		so[i] = typed
+	}
+
+	return so, nil
+}
+
+// GetTLSMutualAuthenticationInput is used as input to the GetTLSMutualAuthentication function.
+type GetTLSMutualAuthenticationInput struct {
+	// ID is an alphanumeric string identifying a mutual authentication (required).
+	ID string
+	// Include is a comma-separated list of related objects to include (optional).
+	Include string
+}
+
+// GetTLSMutualAuthentication retrieves the specified resource.
+func (c *Client) GetTLSMutualAuthentication(i *GetTLSMutualAuthenticationInput) (*TLSMutualAuthentication, error) {
+	if i.ID == "" {
+		return nil, ErrMissingID
+	}
+
+	var ro *RequestOptions
+	if i.Include != "" {
+		ro = &RequestOptions{
+			Params: map[string]string{
+				"include": i.Include,
+			},
+		}
+	}
```

Comment:
Should we set `Accept` here as in the list endpoint?

Reply:
Hmm. I'm not completely sure but I as it's a JSON-API endpoint I don't think it would hurt to have the Accept header set.